packages/generator/tests/acceptance/test_client_generator.py

```diff
@@ -3,7 +3,9 @@
 import re
 import sys
 from typing import Generator
+from unittest.mock import Mock, patch
```

Comment:
Mocks are for unit tests, not acceptance tests. Make this a unit test.

Reply:
I understand that. As, this client generation requires `enumerate services`, `resolve service` and `get metadata` from running measurements. Here, mocking is adapted to set the temp directory as directory out. So, instead of mocking all the services for getting desired path, I have used used `os.chdir` to set temp directory as directory out for testing purpose.

Reply:
You are still using `unittest.mock` to patch the `click.prompt` function.

Read https://click.palletsprojects.com/en/8.1.x/testing/#input-streams for an alternative that is acceptable in acceptance/component tests.

Reply:
I have removed the unittest.mock and updated with `input` to pass the inputs.